pix2tex/dataset/render.py

```diff
@@ -7,112 +8,165 @@
 from tqdm.auto import tqdm
 import cv2
 import numpy as np
+from PIL import Image
+import traceback
+import subprocess
+import shlex
 
 
-def render_dataset(dataset: np.ndarray, names: np.ndarray, args):
-    '''Renders a list of tex equations
+def get_installed_fonts(tex_path: str):
+    cmd = "find %s -name *Math*.otf" % tex_path
+    process = subprocess.Popen(shlex.split(cmd),
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               universal_newlines=True
+                               )
```

Comment:
This only runs for me with `shell=True`

Reply:
What's your OS and python version? Maybe we can add that, not pretty sure about the reason.

---

what I know is

- it works in my WSL2 Ubuntu 20.04.4 LTS
- it works in server Ubuntu 18.04 LTS
---

Here is a tutorial about [python call shell](https://janakiev.com/blog/python-shell-commands/), it does not mention `shell=True`

The reason I delete that line is to consider for [reduce platform-dependent](https://stackoverflow.com/a/3172488/11430711)

---

Here are some code for compare. It seems like  `shlex.split` can replace `shell=True`.
1. without `shlex.split` and `shell=True`
```python
import shlex
import subprocess
cmd = 'echo "More output"'
process = subprocess.Popen(cmd,
                           stdout=subprocess.PIPE,
                           stderr=subprocess.PIPE,
                           universal_newlines=True)
```
output
```shell
FileNotFoundError: [Errno 2] No such file or directory: 'echo "More output"'
```
2. without `shlex.split` and  with `shell=True`
```python
import shlex
import subprocess
cmd = 'echo "More output"'
process = subprocess.Popen(cmd,
                           shell=True,
                           stdout=subprocess.PIPE,
                           stderr=subprocess.PIPE,
                           universal_newlines=True)
stdout, stderr = process.communicate()
print(stdout, stderr, process.returncode)
```
output
```shell
More output
  0
```
3. with`shlex.split` but without `shell=True`
```python
import shlex
import subprocess
cmd = 'echo "More output"'
process = subprocess.Popen(shlex.split(cmd),
                           stdout=subprocess.PIPE,
                           stderr=subprocess.PIPE,
                           universal_newlines=True)
stdout, stderr = process.communicate()
print(stdout, stderr, process.returncode)
```
output
```shell
More output
  0
```

Reply:
On a similar setup but executed from the windows cmd. It throws an error `File *Math*.otf not found`. With `shell=True` it works. So remove `shlex.split` and add `shell=True` sounds like a good tradeoff.
It won't work on stock windows either way.

Reply:
Agree with you